gpflow/kernels/base.py

```diff
@@ -117,6 +120,19 @@ def slice_cov(self, cov: tf.Tensor) -> tf.Tensor:
 
         return cov
 
+    def validate_ard_parameter(self, X, parameter):
+        """
+        Check that the input dimension matches the dimension of the parameter.
+        """
+        if tf.rank(parameter) == 0:
+            return
+        X_shape = tf.shape(X)
+        l_shape = tf.shape(parameter)
+        if l_shape[-1] != X_shape[-1]:
+            # TODO: Maybe this should be a tf.errors.InvalidArgumentError
```

Comment:
I don't think we should use another package's exceptions, especially if they're that unspecific...

Reply:
Agreed - will remove comment